libsolidity/formal/SMTEncoder.cpp

```diff
@@ -953,18 +953,21 @@ pair<smt::Expression, smt::Expression> SMTEncoder::arithmeticOperation(
 
 	smt::Expression intValueRange = (0 - smt::minValue(intType)) + smt::maxValue(intType) + 1;
 	auto value = smt::Expression::ite(
-		valueNoMod > smt::maxValue(intType) || valueNoMod < smt::minValue(intType),
+		valueNoMod > smt::maxValue(intType),
 		valueNoMod % intValueRange,
-		valueNoMod
+		smt::Expression::ite(
+			valueNoMod < smt::minValue(intType),
+			valueNoMod % intValueRange,
+			valueNoMod
+		)
```

Comment:
@ekpyron Split this in this PR also because Z3 folks told me this would simplify things and maybe make it easier for loops.